save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/LnkUserOrganizationController.kt

```diff
@@ -109,16 +107,18 @@ class LnkUserOrganizationController(
     fun getRole(
         @PathVariable organizationName: String,
         @RequestParam(required = false) userName: String?,
-        authentication: Authentication,
-    ): Mono<Role> = getUserAndOrganizationWithPermissions(
-        userName ?: authentication.toUser().name!!,
-        organizationName,
-        Permission.READ,
-        authentication,
-    )
-        .map { (user, organization) ->
-            lnkUserOrganizationService.getRole(user, organization)
-        }
+        authentication: Authentication?,
+    ): Mono<Role> = authentication?.let {
+        getUserAndOrganizationWithPermissions(
+            userName ?: authentication.toUser().name!!,
```

Comment:
`userName` can be provided but `authentication` can be `null` and it still leads to `Role.NONE`
```suggestion
        authentication: Authentication?,
    ): Mono<Role> = (userName ?: authentication?.toUser()?.name)?.let { _userName ->
        getUserAndOrganizationWithPermissions(
            _userName,
            organizationName,
            Permission.READ,
            authentication,
        )
            .map { (user, organization) ->
                lnkUserOrganizationService.getRole(user, organization)
            }
    } ?: Role.NONE.toMono()
```

Reply:
Can we have a UserName for non-authenticated users? After all, we take this name just from the database after authentication

Reply:
userName param is _required = false_. Somewhere in the code we use userName to call the controller, but somewhere we don’t pass anything and use authentication. I don't know why it was implemented like this. But if you remove _authentication_, then the logic will break.

Reply:
Removed UserName